File: Year 1/Term 2/Programming/C++/Lab 2/Project1/Project1/countwinter.cpp
#include "countwinter.h"

int count_winter(string name) {
	int count = 0;
	ifstream file_read;
	file_read.open(name, ios::binary);

	if (!file_read.is_open()) {
		cout << "\n__________Файл " << name << " не вдалося відкрити для підрахунку творів, написаних узимку__________\n";
	}
	else {
		cout << "\n__________Файл " << name << " відкрито для підрахунку творів, написаних узимку__________\n";
		Book temp;
		while (file_read.read((char*)&temp, sizeof(Book))) {
			string date = temp.date;
			date = date.substr(3, 2);
			if (date == "12" || date == "01" || date == "02") {
				count++;
			}
		}

		cout << "\n__________Файл " << name << " закрито__________\n";
	}

	file_read.close();
	return count;
}

File: Year 1/Term 2/Programming/Python/Lab 2/countwinter.py
import pickle
import Book


def count_winter(name: str):
    with open(name, "rb") as file_read:
        print("\n__________Файл " + str(name) + " відкрито для підрахунку творів, написаних узимку__________\n")
        array = pickle.load(file_read)
        count = 0
        for book in array:
            date = book.date[3:]
            if date == "12" or date == "01" or date == "02":
                count += 1
        print("\n__________Файл " + str(name) + " закрито для підрахунку творів, написаних узимку__________\n")
    return count
